py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,512 @@
+        (default 300)
+
+Format placeholders:
+    {events} All the events to display.
+
+Format_event & format_event_full placeholders:
+    {description} The description for the calendar event.
+    {end_date} The end date for the event.
+    {end_time} The end time for the event.
+    {location} The location for the event.
+    {start_date} The start date for the event.
+    {start_time} The start time for the event.
+    {summary} The summary (i.e. title) for the event.
+    {time_to} The time until the event starts (or until it is over
+        if already in progress).
+
+Format_time_to and format_time_left placeholders:
+    {days} The number of days until the event.
+    {hours} The number of hours until the event.
+    {minutes} The number of minutes until the event.
+
+Color options:
+    color_separator: Color for separator.
+    color_event: Color for a single event.
+    color_time: Color for the time associated with each event.
+
+Requires:
+    1. Python library google-api-python-client (can be obtained with pip).
+    2. OAuth 2.0 credentials for the Google Calendar api.
+
+    Follow Step 1 of the guide here to obtain your OAuth 2.0 credentials:
+    https://developers.google.com/google-apps/calendar/quickstart/python
+
+    Download the client_secret.json file which contains your client ID and
+    client secret. In your i3status config file, set configuration parameter
+    client_secret to the path to your client_secret.json file.
+
+    The first time you run the module, a browser window will open asking you
+    to authorize access to your calendar. After authorization is complete,
+    an access/refresh token will be saved to the path configured in
+    auth_token, and i3status will be restarted. This restart will
+    occur only once after the first time you successfully authorize.
+
+Examples:
+```
+# add color gradients for events and dates/times
+google_calendar {
+    thresholds = {
+        'event': [(1, '#d0e6ff'), (2, '#bbdaff'), (3, '#99c7ff'), (4, '#86bcff'),
+            (5, '#62a9ff'), (6, '#8c8cff'), (7, '#7979ff')],
+        'time': [(1, '#ffcece'), (2, '#ffbfbf'), (3, '#ff9f9f'), (4, '#ff7f7f'),
+            (5, '#ff5f5f'), (6, '#ff3f3f'), (7, '#ff1f1f')]
+    }
+}
+```
+
+@author Igor Grebenkov
+@license BSD
+
+SAMPLE OUTPUT
+[
+    {
+      'full_text':'Homer's Birthday (742 Evergreen Terrace) (1h 23m)',
+      'index':0,
+      'separator':False,
+      'separator_block_width':0,
+      'cached_until':1509008337.0
+   },
+   {
+      'full_text':'|',
+      'index':'sep',
+      'separator':False,
+      'separator_block_width':0
+   },
+   {
+      'full_text':'Doctor's Appointment',
+      'index':1,
+      'separator':False,
+      'separator_block_width':0
+   },
+   {
+      'full_text':'|',
+      'index':'sep',
+      'separator':False,
+      'separator_block_width':0
+   },
+   {
+      'full_text':'Lunch with John',
+      'index':2
```

Comment:
We don't need `index` here. Probably best if we wait until right before the merge.

Reply:
This isn't something I did wrong, right? It's just how the module outputs?